### PR TITLE
Small rclc lifecycle tutorial

### PR DESCRIPTION
**Pull request by norro**
Shows how to set up an rclc lifecycle node, transition through its lifecycle states, and assign callbacks to lifecycle transitions.

Fixes issue #179 as part of #143 .

### CODE REVIEW

File: _docs/tutorials/core/programming_rcl_rclc/index.md [hunk 1]
```diff
@@ -213,34 +213,121 @@ if (rc != RCL_RET_OK) {
 }
 ```
 
+## <a name="lifecycle"/>Lifecycle
+
+The rclc lifecycle package provides convenience functions in C to bundle an rclc node with the ROS 2 Node Lifecycle state machine, similar to the [rclcpp Lifecycle Node](https://github.com/ros2/rclcpp/blob/master/rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_node.hpp) for C++.
```
Comment (ralph-lange) [inline _docs/tutorials/core/programming_rcl_rclc/index.md:218]:
```suggestion
The rclc lifecycle package provides convenience functions in C to bundle an rcl node with the ROS 2 Node Lifecycle state machine, similar to the [rclcpp Lifecycle Node](https://github.com/ros2/rclcpp/blob/master/rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_node.hpp) for C++.
```

File: _docs/tutorials/core/programming_rcl_rclc/index.md [hunk 2]
```diff
@@ -213,34 +213,121 @@ if (rc != RCL_RET_OK) {
...
+
+This tutorial show-cases how to set up an rclc lifecycle node, transition through its lifecycle states, and assign callbacks to lifecycle transitions.
```
Comment (ralph-lange) [inline _docs/tutorials/core/programming_rcl_rclc/index.md:220]: We may also name it `lifecycle-enabled rcl node`, but I have no string opinion on that.

File: _docs/tutorials/core/programming_rcl_rclc/index.md [hunk 3]
```diff
@@ -213,34 +213,121 @@ if (rc != RCL_RET_OK) {
...
+
+### Initialization
+
+Creation of a lifecycle node as a bundle of an rcl node and the rcl Node Lifecycle state machine.
```
Comment (ralph-lange) [inline _docs/tutorials/core/programming_rcl_rclc/index.md:224]:
```suggestion
Creation of a lifecycle node as a bundle of an rcl node and the rcl lifecycle state machine.
```
I would omit Node for readability.

File: _docs/tutorials/core/programming_rcl_rclc/index.md [hunk 4]
```diff
@@ -213,34 +213,121 @@ if (rc != RCL_RET_OK) {
...
+rclc_support_t support;
+rcl_ret_t rc;
+
+// create rcl node
+rc = rclc_support_init(&support, argc, argv, &allocator);
+rcl_node_t my_node = rcl_get_zero_initialized_node();
+rc = rclc_node_init_default(&my_node, "lifecycle_node", "rclc", &support);
+
+// rcl state machine
+rcl_lifecycle_state_machine_t state_machine_ =
```
Comment (ralph-lange) [inline _docs/tutorials/core/programming_rcl_rclc/index.md:239]:
```suggestion
rcl_lifecycle_state_machine_t state_machine =
```

File: _docs/tutorials/core/programming_rcl_rclc/index.md [hunk 5]
```diff
@@ -213,34 +213,121 @@ if (rc != RCL_RET_OK) {
...
+  rcl_lifecycle_get_zero_initialized_state_machine();
+...
+
+// create the lifecycle node
+rclc_lifecycle_node_t lifecycle_node;
+rcl_ret_t rc = rclc_make_node_a_lifecycle_node(
+  &lifecycle_node,
+  &my_node,
+  &state_machine_,
```
Comment (ralph-lange) [inline _docs/tutorials/core/programming_rcl_rclc/index.md:248]:
```suggestion
  &state_machine,
```

File: _docs/tutorials/core/programming_rcl_rclc/index.md [hunk 6]
```diff
@@ -213,34 +213,121 @@ if (rc != RCL_RET_OK) {
...
+rcl_node_t my_node = rcl_get_zero_initialized_node();
+rc = rclc_node_init_default(&my_node, "lifecycle_node", "rclc", &support);
+
+// rcl state machine
+rcl_lifecycle_state_machine_t state_machine_ =
+  rcl_lifecycle_get_zero_initialized_state_machine();
+...
+
+// create the lifecycle node
+rclc_lifecycle_node_t lifecycle_node;
```
Comment (ralph-lange) [inline _docs/tutorials/core/programming_rcl_rclc/index.md:244]: Should it be named `my_lifecycle_node` for consistency?

File: _docs/tutorials/core/programming_rcl_rclc/index.md [hunk 7]
```diff
@@ -213,34 +213,121 @@ if (rc != RCL_RET_OK) {
...
+
+To clean everything up, simply do
+
+```C
+rc += rcl_lifecycle_node_fini(&lifecycle_node, &allocator);
+```
+
+### Example and Limitations
+
+An example, how to use the RCLC Lifecycle Node is given in the file `lifecycle_node.c` in the [rclc_examples](https://github.com/micro-ROS/rclc/tree/master/rclc_examples) package.
```
Comment (ralph-lange) [inline _docs/tutorials/core/programming_rcl_rclc/index.md:295]:
```suggestion
An example, how to use the rclc Lifecycle Node is given in the file `lifecycle_node.c` in the [rclc_examples](https://github.com/micro-ROS/rclc/tree/master/rclc_examples) package.
```

File: _docs/tutorials/core/programming_rcl_rclc/index.md [hunk 8]
```diff
@@ -213,34 +213,121 @@ if (rc != RCL_RET_OK) {
...
+
+The state machine publishes state changes, however, lifecycle services are not yet exposed via ROS 2 services (tbd).
```
Comment (ralph-lange) [inline _docs/tutorials/core/programming_rcl_rclc/index.md:297]: I propose to open an issue on that at ros2/rcl and notify the original lifecycle developer on it.

Comment (norro) [reply _docs/tutorials/core/programming_rcl_rclc/index.md:297]: I think this is actually intended. The services need to be set in the context of the node, so our statemachine-node-bundle seems to be a good place for it. See https://github.com/micro-ROS/rclc/issues/40

File: _docs/tutorials/core/programming_rcl_rclc/index.md (hunk 2 above)
Comment (norro) [reply _docs/tutorials/core/programming_rcl_rclc/index.md:220]: I think I leave it like it is for now to be consistent with the `rclcpp LifecycleNode`.

**Pull request merged**